pegen/pegen.c

```diff
@@ -505,22 +505,22 @@ seq_flatten(Parser *p, asdl_seq *seqs)
 }
 
 expr_ty
-seq_to_dotted_name(Parser *p, asdl_seq *seq)
+join_names_with_dot(Parser *p, expr_ty first_name, expr_ty second_name)
 {
-    /* Create a string of the form a.b.c like Python/ast.c alias_for_dotted_name does
-       for dotted names */
-    if (asdl_seq_LEN(seq) == 1) {
-        return asdl_seq_GET(seq, 0);
-    }
+    /* Create a string of the form "first_name.second_name" like
+       Python/ast.c alias_for_dotted_name does for dotted names */
+    PyObject *first_identifier = first_name->v.Name.id;
+    PyObject *second_identifier = second_name->v.Name.id;
 
-    // Get length of dotted string
-    ssize_t len = 0;
-    for (int i = 0, l = asdl_seq_LEN(seq); i < l; i++) {
-        expr_ty current_expr = asdl_seq_GET(seq, i);
-        ssize_t current_len = PyUnicode_GET_LENGTH(current_expr->v.Name.id);
-        len += current_len + 1;
+    if (PyUnicode_READY(first_identifier) == -1) {
+        return NULL;
+    }
+    if (PyUnicode_READY(second_identifier) == -1) {
+        return NULL;
     }
-    --len; // Last name does not have a dot
+    ssize_t len = PyUnicode_GET_LENGTH(first_name->v.Name.id) +
+                  PyUnicode_GET_LENGTH(second_name->v.Name.id) + 1; // +1 for the dot
```

Comment:
You should really calculate the length using `strlen(first_str) + 1 + strlen(second_str)`, then you get it right. And move the code around of course.